src/rpc/blockchain.cpp

```diff
@@ -287,10 +288,26 @@ static RPCHelpMan waitfornewblock()
     NodeContext& node = EnsureAnyNodeContext(request.context);
     Mining& miner = EnsureMining(node);
 
-    // Abort if RPC came out of warmup too early
+    /**
+     * If the caller provided a current_tip value, pass it to waitTipChanged().
+     *
+     * If the caller did not provide a current tip hash, call getTip() to get
+     * one and wait for the tip to be different from this value. This mode is
+     * less reliable because if the tip changed between waitfornewblock calls,
+     * it will need to change a second time before this call returns.
+     *
+     * Abort if RPC came out of warmup too early.
```

Comment:
In commit "rpc: add optional blockhash to waitfornewblock" (0625f26b09084fcbd4daa48b3456f54eff6b22de)

Do we think "Abort if RPC came out of warmup too early." comment is still accurate? I've having trouble understanding what it could mean and I don't think there are cases where this aborted even before this change. Would probably suggest dropping this comment or trying to make it clearer.

Also could replace /** with a normal comment above since /** normally indicates a doxygen comment.

Reply:
I replaced the `/**`.

I dropped the warmup comment, but will keep the `CHECK_NONFATAL`. If the RPC starts listening before `ActiveChain().Tip()` is set, then `getTip()` returns nothing. I'm not sure if that can ever happen. `SetRPCWarmupFinished()` is called as step 13 in `init.cpp` with a somewhat ominous comment above it.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30635#discussion_r2052789582

Thanks, I think I get it now. "Abort if RPC came out of warmup too early" meant "There is a bug if RPC came out of warmup too early, so abort". But it should be pretty obvious that if a check statement fails, there is a bug, so the comment seems redundant and your change dropping the comment and keeping the check makes sense.